CMakeLists.txt

```diff
@@ -157,6 +157,17 @@ if (OGS_USE_EIGENLIS)
     add_definitions(-DOGS_USE_EIGENLIS)
 endif()
 
+if(OGS_BUILD_UTILS)
+    option(UTILS_PARTMESH_BUILD_WITH_METIS "Compile partmesh source code together with METIS ones." OFF)
+
+    if(UTILS_PARTMESH_BUILD_WITH_METIS)
+        set(METIS_PATH "${CMAKE_SOURCE_DIR}/ThirdParty/metis")
+        add_definitions(-DBUILD_WITH_METIS)
+        include(${CMAKE_SOURCE_DIR}/Applications/Utils/ModelPreparation/PartitionMesh/MetisSetup.cmake)
+    endif()
+endif()
```

Comment:
Maybe this part should be more general and not be bound to a specific tool. Something like this:

If metis submodule was checked out then

- include `MetisSetup.cmake`
- include `ModelPreparation/PartitionMesh` (in Utils-directory)